src/compilation/compiler.rs

```diff
@@ -258,6 +304,7 @@ impl Compiler {
         }
     }
 
+    /// Compiles a statement, assuming the LLVM builder is building inside a function body.
```

Comment:
Good catch. There is no check that LLVM builder is inside a function anywhere. We have a function called get_cur_function() that is supposed to get the current function BUT it can actually return null. We should be checking in the get_cur_function that the return value is not null. Because we later inthe code use the return value of that function assuming that it's not null. But if the builder is not within a function body, god knows what kinda of null pointer dereferncing might happen :)

Reply:
Very good point lol. I updated the get_cur_function() to panic if the current function is NULL (3917bf1).

The good thing is that get_cur_function() is only ever used by us. So, as long as we're careful to only call it from places when the builder is inside a function, we're good. And if we aren't, we'll get a nice panic message.